### PR TITLE
Add REST API for application token generation

### PR DESCRIPTION
**Pull request by ruwiniwj**
## Purpose
Add REST API for application token generation
Fix https://github.com/wso2/product-apim/issues/4816

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml [hunk 1]
```diff
@@ -3780,14 +3780,19 @@ definitions:
...
-        type: string
+        type: array
+        items:
+          type: string
         description: Allowed scopes (space seperated) for the access token
       revokeToken:
         type: string
-        description: Token to be revoked, if any.
+        description: Token to be revoked, if any
+      jsonInput:
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: shall we rename this additionalProperties?

Comment (ruwiniwj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml [hunk 2]
```diff
@@ -3780,14 +3780,19 @@ definitions:
         type: string
         description: Consumer secret of the application
       validityPeriod:
-        type: integer
+        type: string
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: Should we make this an String?
If the requirement is to have a Long integer, we can use 
```
type: integer
format: int64
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApplicationsApiServiceImpl.java [hunk 3]
```diff
@@ -408,8 +414,62 @@ public Response applicationsApplicationIdKeysGet(String applicationId, MessageCo
...
+            APIConsumer apiConsumer = RestApiUtil.getConsumer(username);
+            Application application = apiConsumer.getApplicationByUUID(applicationId);
+
+            if (application != null) {
+                if (RestAPIStoreUtils.isUserAccessAllowedForApplication(application)) {
+                    ApplicationKeyDTO appKey = getApplicationKeyByAppIDAndKeyType(applicationId, keyType);
+                    if (appKey != null) {
+                        String jsonInput = null;
+                        try {
+//                          verify that the provided validity period is a long
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApplicationsApiServiceImpl.java]: Should we move the comment //s here to the beginning of the texts?

Comment (ruwiniwj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApplicationsApiServiceImpl.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/6704

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml (hunk 2 above)
Comment (ruwiniwj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: Fixed with https://github.com/wso2/carbon-apimgt/pull/6704

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApplicationsApiServiceImpl.java [hunk 4]
```diff
@@ -423,12 +483,23 @@ public Response applicationsApplicationIdKeysKeyTypeGenerateTokenPost(String app
...
+     * @param keyType       Key Type (Production | Sandbox)
+     * @return Application Key Information
+     */
+    private ApplicationKeyDTO getApplicationKeyByAppIDAndKeyType(String applicationId, String keyType) {
         List<APIKey> applicationKeys = getApplicationKeys(applicationId);
-        for (APIKey apiKey : applicationKeys) {
-            if (keyType != null && keyType.equals(apiKey.getType())) {
-                ApplicationKeyDTO appKeyDTO = ApplicationKeyMappingUtil.fromApplicationKeyToDTO(apiKey);
-                return Response.ok().entity(appKeyDTO).build();
+        if (applicationKeys!= null) {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApplicationsApiServiceImpl.java]: There is a formatting issue

Comment (ruwiniwj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApplicationsApiServiceImpl.java]: Fixed with https://github.com/wso2/carbon-apimgt/pull/6753

**Pull request merged**